echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py

```diff
@@ -0,0 +1,1091 @@
+import numpy as np
+import xarray as xr
+
+# TODO: turn this into an absolute import!
+from ...core import SONAR_MODELS
+from ..convention import sonarnetcdf_1
+
+_varattrs = sonarnetcdf_1.yaml_dict["variable_and_varattributes"]
+
+
+def _range_bin_to_range_sample(ed_obj):
+    """
+    Renames the coordinate range_bin to range_sample.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    for grp_path in ed_obj.group_paths:
+
+        if "range_bin" in list(ed_obj[grp_path].coords):
+
+            # renames range_bin in the dataset
+            ed_obj[grp_path] = ed_obj[grp_path].rename(name_dict={"range_bin": "range_sample"})
+
+            ed_obj[grp_path].range_sample.attrs["long_name"] = "Along-range sample number, base 0"
+
+
+def _add_attrs_to_freq(ed_obj):
+    """
+    Makes the attributes of the ``frequency`` variable
+    consistent for all groups. This is necessary because
+    not all groups have the same attributes (some are
+    missing them too) for the ``frequency`` variable.
+    This variable is used to set the variable
+    ``frequency_nominal`` later on.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    for grp_path in ed_obj.group_paths:
+
+        if "frequency" in list(ed_obj[grp_path].coords):
+
+            # creates consistent frequency attributes
+            # NOTE: the attribute standard_name will be removed in `frequency_nominal`
+            ed_obj[grp_path]["frequency"] = ed_obj[grp_path].frequency.assign_attrs(
+                {
+                    "long_name": "Transducer frequency",
+                    "standard_name": "sound_frequency",
+                    "units": "Hz",
+                    "valid_min": 0.0,
+                }
+            )
+
+
+def _reorganize_beam_groups(ed_obj):
+    """
+    Maps Beam --> Sonar/Beam_group1 and Beam_power --> Sonar/Beam_group2.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    # Map Beam --> Sonar/Beam_group1
+    if "Beam" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam"])
+        ed_obj._tree["Sonar/Beam"].name = "Beam_group1"
+
+    # Map Beam_power --> Sonar/Beam_group2
+    if "Beam_power" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam_power"])
+        ed_obj._tree["Sonar/Beam_power"].name = "Beam_group2"
+
+
+def get_channel_id(ed_obj, sensor):
+    """
+    Returns the channel_id for all non-unique frequencies.
```

Comment:
Could you elaborate on what you meant by "non-unique" here? Do you mean just all the `channel_id` will be included and concatenated?

Reply:
This function will grab all channel ids, even those channel ids where the frequencies are repeated (that's why I say "non-unique frequencies"). If you have a better sentence to describe this, please let me know.

Reply:
Going back to a comment found elsewhere, there will be no 0.5.x converted files with repeated frequencies. 0.5.x could not generate such files.  So, there's no need to test or account for that case.

Reply:
To clarify a little further, I specifically state this for cases such as the EK80 sensor. When we have two beam groups, we may have frequencies that are the same, but have a different channel id. Note that I am collecting the channel ids from all beam groups (for EK80).

Reply:
I see. I had not considered that case